SI/Sem.07/Pract.07/References/Task07.cpp

```diff
@@ -0,0 +1,125 @@
+#include <iostream>
+
+unsigned const int MAX_COUNT_ROWS = 100;
+unsigned const int MAX_COUNT_COLS = 3;
+bool GetInput(int movies[MAX_COUNT_ROWS][MAX_COUNT_COLS], int countRows);
+void FindEndTime(int movies[MAX_COUNT_ROWS][MAX_COUNT_COLS], int countRows);
+int* ConvertToMinutes(int duration);
+int* SumTime(int hoursBegin, int minutesBegin, int hoursDuration, int minutesDuration);
+bool CheckValidInput(int hours, int minutes, int duration);
+void Print(int hoursEnd, int minutesEnd);
+
+int main()
+{
+	int movies[MAX_COUNT_ROWS][MAX_COUNT_COLS];
+	int countRows = 0;
+	int a, b;
+	std::cin >> countRows;
+	if (!GetInput(movies, countRows))
+	{
+		return 0;
+	}
+	FindEndTime(movies, countRows);
+}
+
+bool GetInput(int movies[MAX_COUNT_ROWS][MAX_COUNT_COLS], int countRows)
+{
+	for (int i = 0; i < countRows; i++)
+	{
+		int hourBegin = 0;
+		int minutesBegin = 0;
+		int duration = 0;
+		std::cin >> hourBegin >> minutesBegin >> duration;
+		if (!CheckValidInput(hourBegin, minutesBegin, duration))
+		{
+			std::cout << "Invalid input!";
+			return false;
+		}
+		else
+		{
+			
+			movies[i][0] = hourBegin;
+			movies[i][1] = minutesBegin;
+			movies[i][2] = duration;
+		}
+	}
+	return true;
+}
+
+bool CheckValidInput(int hours, int minutes, int duration)
+{
+	if (hours >= 0 && minutes >= 0 && duration >= 0)
+	{
+		return true;
+	}
+	return false;
+}
+
+void FindEndTime(int movies[MAX_COUNT_ROWS][MAX_COUNT_COLS], int countRows)
+{
+	int maxLatestHour = -1;
+	int maxLatestMinutes = -1;
+
+	for (int i = 0; i < countRows; i++)
+	{
+		int duration = movies[i][2];
+		int hoursOfDuration = ConvertToMinutes(duration)[0];
+		int minutesOfDuration = ConvertToMinutes(duration)[1];
+
+		int endHour = SumTime(movies[i][0], movies[i][1], hoursOfDuration, minutesOfDuration)[0];
+		int endMinutes = SumTime(movies[i][0], movies[i][1], hoursOfDuration, minutesOfDuration)[1];
```

Comment:
@Justsvetoslavov Why these two lines work correctly?

Reply:
+ you are calculating the same thing two times

Reply:
@GeorgiTerziev02 We have reworked the solution in class.

As for the above question, it is returning a pointer and we are referencing by offset afterwards.(really hacky which we removed)